streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala

```diff
@@ -157,7 +157,7 @@ class FileInputDStream[K, V, F <: NewInputFormat[K, V]](
     val metadata = Map(
       "files" -> newFiles.toList,
       StreamInputInfo.METADATA_KEY_DESCRIPTION -> newFiles.mkString("\n"))
-    val inputInfo = StreamInputInfo(id, 0, metadata)
+    val inputInfo = StreamInputInfo(id, rdds.map(_.count).sum, metadata)
```

Comment:
This will kick off a new Spark job to read files and count, which will bring in obvious overhead. Whereas `count` in `DirectKafkaInputDStream` only calculates offsets.

Reply:
Because of this little overhead, that 'Records' is not recorded? This is a obvious bug.

Reply:
I see what you mean. I'll try to make it read once. Can you give me some idea?

Reply:
I'm not sure if there's a solution to fix it here.

Reply:
I would like another from a thread, try hdfs api to count the number of documents, does not affect the main thread spark jobs, i will try. ok? @jerryshao

Reply:
What actually do you want, number of files or number of records?

Reply:
The number of rows in a file. Is this solution possible?

Reply:
What's the difference between using HDFS API to count in another thread compared to current solution? You still cannot avoid reading files twice.

Reply:
I'm not in favor of such changes. No matter the process is sync or async, because `reportInfo` is invoked here, so you have to wait for the process to end.

Anyway I think reading twice is unacceptable for streaming scenario (even for batch scenario). I guess the previous code set to "0" by intention.

Reply:
If you can add a switch parameter, the default value is false.

If it is true, then it needs to be count (read the file again) so that the records can be correctly counted. Of course, it shows that when the parameter is opened to true, the streaming performance problem will be affected.

Reply:
I don't think it's a good idea. Actually I'm incline of leaving as it is.